### ISSUE TITLE
Contributions to context menu

### ISSUE THREAD

**Issue by shroffk** [ControlSystemStudio/phoebus#1374]
In the past we have had request to allow configuring the addition and removal of context menu entires. This was achieved in the issue ticket listed below by preferences.
#1094

The use of preferences still requires building and packaging of modules not really needed by the product... with the preference basically excluding a module that was included. 
This pattern of including plugins also results in other problems, like warnings from the channelfinder plugin on startup which are very important at a site which has channelfinder but can't access it and are completely irrelevant at a site which has no channelfinder.

To address both these the phoebus framework should enable the easy creation of products with just the right pieces to address the needs of a site without forcing the inclusion of superflusious peices.

**Comment by shroffk** [comment]
In order to achive this here are a few things.

1. The selection, adapter, and contextmenu API's are services can be improved and simplifed.

By looking at existing context menu contributions I have updated the selection API, I have added a few more methods and made most of the call method default.
Additionally thanks to the improvement to the contenxtmenu service by @kasemir and updates to adapter and selection service a simpler design pattern for making site specifc contributions to the context menu is significantly reduced.

The trends module with 2 implementation for contributing the email action to the databrowser can be used as an example.

The default/simple email action
![image](https://user-images.githubusercontent.com/2111304/83760465-f9f72f00-a642-11ea-86a9-bf3b5fc65abf.png)

The rich email action, this includes the list of pvs, the timerange, and a copy of the .plt file
![email-rich](https://user-images.githubusercontent.com/2111304/83760474-fcf21f80-a642-11ea-86ee-5feda450ee1d.PNG)


Sites which like the default can stick to the simple email action.
Sites which like the rich email options can inlcude the app-trends-rich-context-menu in their product.
Sites which want no email action can not include anything.
